xcdat/spatial_avg.py

```diff
@@ -634,10 +695,7 @@ def _combine_weights(
             ``weights`` are 1-D and correspond to the specified axes (``axis``)
             in the region.
         """
-        weights = {
-            axes: weights for axes, weights in axis_weights.items() if axes in axis
-        }
```

Comment:
No longer need to parse the `axis_weights` dict because `get_weights()` will only calculate weights based on `axis` arg